pkg/validatePoliciesYaml/validatePoliciesYaml.go

```diff
@@ -30,5 +34,128 @@ func ValidatePoliciesYaml(content []byte, policyYamlPath string) error {
 		return validationErrors
 	}
 
+	return validatePoliciesContent(jsonContent)
+}
+
+func validatePoliciesContent(content []byte) error {
+	// unmarshal the content
+	var schema *cliClient.EvaluationPrerunPolicies
+	err := json.Unmarshal(content, &schema)
+	if err != nil {
+		return err
+	}
+
+	// validate that exactly one policy is set to default
+	err = validateSingleDefaultPolicy(schema.Policies)
+	if err != nil {
+		return err
+	}
+
+	// validate if the policy file has any identifier related issues
+	err = validateIdentifier(schema.Policies, schema.CustomRules)
+	if err != nil {
+		return err
+	}
+
+	// validate the schema of each rule
+	err = validateSchemaField(schema.CustomRules)
+	return err
+}
+
+func validateIdentifier(policies []*cliClient.Policy, customRules []*cliClient.CustomRule) error {
+
+	err := checkIdentifierUniquenessInPolicy(policies)
+	if err != nil {
+		return err
+	}
+
+	err = checkIdentifierUniqueness(customRules)
+	return err
+}
+
+func checkIdentifierUniquenessInPolicy(policies []*cliClient.Policy) error {
+	for _, policy := range policies {
+		propertyValuesExistenceMap := make(map[string]bool)
+		rules := policy.Rules
+		for _, rule := range rules {
+			identifier := rule.Identifier
+			if propertyValuesExistenceMap[identifier] {
+				return fmt.Errorf("invalid policy file: identifier \"%s\" used atleast two times in policy \"%s\"", identifier, policy.Name)
+			}
+			propertyValuesExistenceMap[identifier] = true
+		}
+	}
+	return nil
+}
+
+func checkIdentifierUniqueness(customRules []*cliClient.CustomRule) error {
+	defaultRules, err := defaultRules.GetDefaultRules()
+	if err != nil {
+		return err
+	}
+	propertyValuesExistenceMap := make(map[string]bool)
+
+	for _, item := range customRules {
+		identifier := item.Identifier
+
+		if propertyValuesExistenceMap[identifier] {
+			return fmt.Errorf("invalid policy file: identifier \"%s\" is used in atleast two rules", identifier)
+		}
+
+		propertyValuesExistenceMap[identifier] = true
+	}
+
+	for _, item := range defaultRules.Rules {
+		identifier := item.UniqueName
+
+		if propertyValuesExistenceMap[identifier] {
+			return fmt.Errorf("invalid policy file: a default rule with same identifier \"%s\" already exists", identifier)
```

Comment:
please match this error message to match the updated structure of the other error messages and consider adding a test case :)